x-pack/plugins/canvas/public/services/expressions.ts

```diff
@@ -32,9 +37,15 @@ export const expressionsServiceFactory: CanvasServiceFactory<ExpressionsService>
         // function that matches its definition, but which simply
         // calls the server-side function endpoint.
         Object.keys(serverFunctionList).forEach((functionName) => {
-          if (expressions.getFunction(functionName)) {
+          // Allow function to be overwritten if we want to use
+          // the server-hosted essql, esdocs, and escount functions
+          if (
+            !(!hasDataSearch && dataSearchFns.includes(functionName)) &&
+            expressions.getFunction(functionName)
```

Comment:
This is where the magic happens. On the server, we still load up esql, escount, and esdocs. We load up those three functions (using the data search stuff) on client as well. If we have the data search experiment enabled, we will skip the function and let the client-side version run. If the experiment is disabled, we overwrite the new client version of the function with the function that calls out to the server

Reply:
as a side note, i'm gonna break this if-statement into an if-else for readability. It's too hard to parse at the moment

Reply:
Rock ON, man.  Epic... I like this.